cmd/snap/cmd_set_test.go

```diff
@@ -127,6 +127,20 @@ func (s *snapSetSuite) TestSnapSetFailParsingWithStrictJSON(c *check.C) {
 	c.Assert(err, check.ErrorMatches, "failed to parse JSON:.*")
 }
 
+func (s *snapSetSuite) TestSnapSetFailJsonKeysValidation(c *check.C) {
+	invalidJson := []string{
+		`key={"key0":"val0","key1":"val1","key2":{"key3":"val2","key4":"val3","key5":{"INVALID":"val4","key7":"val5"}},"key8":{"key9":{"key10":"val6"}}}`,
+		`key={"key0":"val0","key1":"val1","key2":{"key3":"val2","key4":"val3",` +
+			`"key5":{"key6":"val4","key7":"val5"}},"key8":{"key9":{"key10":"val6"}},` +
+			`"key11":[{"key12":"val7"},{"KEY13":"val8"},{"key14":["val9","val10"]}]}`,
+	}
+
+	for _, json := range invalidJson {
+		_, err := snapset.Parser(snapset.Client()).ParseArgs([]string{"set", "snapname", "-t", json})
+		c.Assert(err, check.ErrorMatches, "invalid configuration key: invalid option name:.*")
```

Comment:
Nitpick on this error message, it's just weird. See my other comments.

Reply:
Solved! Thanks for your suggestion!